client_authentication.go

```diff
@@ -200,7 +203,10 @@ func (f *Fosite) AuthenticateClient(ctx context.Context, r *http.Request, form u
 	}
 
 	// Enforce client authentication
+	span, _ := opentracing.StartSpanFromContext(ctx, "hash_compare")
```

Comment:
ℹ️ btw, the second returned value here [is not an error](https://github.com/opentracing/opentracing-go/blob/master/gocontext.go#L43), it is the context object that was built around the created span - which you will use to propagate further if you need.

I am being responsible! 😅